pallets/asset-manager/src/lib.rs

```diff
@@ -394,6 +456,89 @@ pub mod pallet {
 		pub fn account_id() -> T::AccountId {
 			T::PalletId::get().into_account()
 		}
+
+		/// Get para id from multilocation
+		pub(crate) fn get_para_id_from_multilocation(
+			location: Option<&MultiLocation>,
+		) -> Option<ParaId> {
+			if let Some(MultiLocation { interior, .. }) = location {
+				match interior {
+					// We have some locations like (1, X1(Parachain)).
+					Junctions::X1(Junction::Parachain(para_id)) => Some(*para_id),
+					Junctions::X2(Junction::Parachain(para_id), Junction::AccountId32 { .. }) => {
+						Some(*para_id)
+					}
```

Comment:
This is the only dest that you need to match against. Don't forget this is the destination account location, not an asset location

Reply:
I just use one function to handle asset location and dest location, will split it.